src/data/pfun.lean

```diff
@@ -448,4 +448,28 @@ ext $ λ _ _, by simp only [comp_apply, part.bind_comp]
 lemma coe_comp (g : β → γ) (f : α → β) : ((g ∘ f : α → γ) : α →. γ) = (g : β →. γ).comp f :=
 ext $ λ _ _, by simp only [coe_val, comp_apply, part.bind_some]
 
+/-- Product of partial functions. -/
+protected def prod (f : α →. γ) (g : β →. δ) : α × β →. γ × δ :=
```

Comment:
It might be nice to also have a lift `(f : α →. γ) (g : α →. δ)` to `α →. γ × δ`, then your definition is a composition of this with the total functions fst/snd:
```
protected def prod_lift (f : α →. β) (g : α →. γ) : α →. β × γ :=
```
I also don't think this should be called `prod` - the name for this sort of function elsewhere is something like `prod_map`

Reply:
I defined `prod_lift`, but didn't redefine `prod_map` using it because it would introduce a stray `true ∧` in `dom`.

Reply:
Okay, can you have the lemma saying they're the same in that case?